output/csv/output_test.go

```diff
@@ -315,161 +265,47 @@ func TestRunCollect(t *testing.T) {
 					Metric: stats.New("my_metric", stats.Gauge),
 					Value:  1,
 					Tags: stats.NewSampleTags(map[string]string{
-						"tag1": "val1",
-						"tag2": "val2",
-						"tag3": "val3",
+						"check": "val1",
+						"url":   "val2",
+						"error": "val3",
 					}),
 				},
 				stats.Sample{
 					Time:   time.Unix(1562324644, 0),
 					Metric: stats.New("my_metric", stats.Gauge),
 					Value:  1,
 					Tags: stats.NewSampleTags(map[string]string{
-						"tag1": "val1",
-						"tag2": "val2",
-						"tag3": "val3",
-						"tag4": "val4",
+						"check": "val1",
+						"url":   "val2",
+						"error": "val3",
+						"name":  "val4",
 					}),
 				},
 			},
 			fileName:       "test.gz",
 			fileReaderFunc: readCompressedFile,
-			outputContent:  "metric_name,timestamp,metric_value,tag1,tag3,extra_tags\n" + "my_metric,1562324643,1.000000,val1,val3,\n" + "my_metric,1562324644,1.000000,val1,val3,tag4=val4\n",
+			outputContent:  "metric_name,timestamp,metric_value,check,error,extra_tags\n" + "my_metric,1562324643,1.000000,val1,val3,url=val2\n" + "my_metric,1562324644,1.000000,val1,val3,name=val4&url=val2\n",
 		},
 	}
 
 	for _, data := range testData {
 		mem := afero.NewMemMapFs()
-		collector, err := New(
-			testutils.NewLogger(t),
-			mem,
-			stats.TagSet{"tag1": true, "tag2": false, "tag3": true},
-			Config{FileName: null.StringFrom(data.fileName), SaveInterval: types.NewNullDuration(time.Duration(1), true)},
-		)
-		assert.NoError(t, err)
-		assert.NotNil(t, collector)
-
-		ctx, cancel := context.WithCancel(context.Background())
-
-		var wg sync.WaitGroup
-		wg.Add(1)
-		go func() {
-			collector.Run(ctx)
-			wg.Done()
-		}()
-		err = collector.Init()
+		output, err := newOutput(output.Params{
+			Logger:         testutils.NewLogger(t),
+			FS:             mem,
+			ConfigArgument: data.fileName,
+			ScriptOptions: lib.Options{
+				SystemTags: stats.NewSystemTagSet(stats.TagError | stats.TagCheck),
+			},
+		})
+		require.NoError(t, err)
+		require.NotNil(t, output)
 
-		assert.NoError(t, err)
-		collector.Collect(data.samples)
```

Comment:
Maybe this test should be renamed to `TestRun`, and the existing `TestRun` merged or removed, since it's only a smoke test.